### PR TITLE
Adds an upload definitions method to upload the definitions

### PR DESCRIPTION
**Pull request by shubhang93**

### CODE REVIEW

**Review by michaelklishin — CHANGES_REQUESTED** [review]
This fails tests because no `arguments` provided on the input data, which makes in fail validation on the HTTP API endpoint side.

File: definitions.go [hunk 1]
```diff
@@ -55,3 +58,24 @@ func (c *Client) ListVhostDefinitions(vhost string) (p *ExportedDefinitions, err
 
 	return p, nil
 }
+
+//
+// POST /api/definitions
+//
+
+// UploadDefinitions uploads a set of definitions and returns an error indicating if the operation was a failure
+func (c *Client) UploadDefinitions(p *ExportedDefinitions) error {
```
Comment (michaelklishin) [inline definitions.go:67]: This could follow e.g. `DeclareQueue` and `DeclareExchange` and return a pair of `(res *http.Response, err error)`

Comment (shubhang93) [reply definitions.go:67]: sure thanks for the feedback, will make the changes

**Comment by shubhang93** [comment]
All the tests are passing, can you merge the changes @michaelklishin

**Comment by shubhang93** [comment]
also can we use a docker container, instead of installing rabbitmq-server on a debian OS?

**Comment by michaelklishin** [comment]
Containers make sense for Actions. I don't want to switch to container-based nodes only for local development but would consider a PR that adds support for containers as an option locally, similarly to [how Bunny does it](https://github.com/ruby-amqp/bunny/blob/master/CONTRIBUTING.md).

**Comment by shubhang93** [comment]
This current PR adds support to use containers locally

**Pull request merged**